src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/UWPtoWinAppSDKUpgrade/WinUIPropertiesUpdater.cs

```diff
@@ -85,6 +104,20 @@ public async Task<IUpdaterResult> ApplyAsync(IUpgradeContext context, ImmutableA
                 new List<string>());
         }
 
+        private string ParseProjectNameWithExtension(string projectReference, string extension)
+        {
+            var index = projectReference.IndexOf(".vcxproj");
+            index--;
+            StringBuilder sb = new StringBuilder();
```

Comment:
I was initially confused by what you're doing here, but I _think_ we can avoid the looping, reversing, etc. by simply doing:
```suggestion
            return new string(projectReference.Substring(0, index).Where(c=>char.IsLetterOrDigit(c)));
```

Reply:
This doesn't solve the purpose. projectReference looks something like this: "`(ProjectId, #82408bc4-5748-4ae3-b1bb-9a6c0bbc3ec2 - ..\\RuntimeComponent1\\RuntimeComponent1.vcxproj)"`
I want to parse the project name which is the part of the path just before `".vcxproj"` which I do character by character until I reach the first non alphanumeric character. Since this is parsed in reverse, I have to reverse the string again.

Reply:
ah what about utilizing `.LastIndexOf(System.IO.Path.PathSeparator)` in this case?